### PR TITLE
Exclude dependabot from changelog.md check

### PR DESCRIPTION
**Pull request by danielkesselberg**
## Description:

We decided to use https://dependabot.com for dependency updates. dependabot is not able to add a new line to changelog.md. Its required to exclude pull requests from dependabot from changelog.md check.

## Improvements
- Early exit when not a pull request
- https://www.shellcheck.net/ is happy

## Known Issues:
_NONE_

### CODE REVIEW

**Comment by aeimer** [comment]
@danielkesselberg should work. I have some little adaption I would like to see fixed. Give me time until tomorrow

File: .travis/changelog.sh [hunk 1]
```diff
@@ -2,43 +2,54 @@
 
 # Path to this file
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-# Path the script was called from
-IPWD="$(pwd)"
```
Comment (PatrickSkowronek) [inline .travis/changelog.sh]: We have a convention on the shell files. All files define the pwd.

File: .travis/changelog.sh [hunk 2]
```diff
@@ -2,43 +2,54 @@
...
-# Import shared vars
-. ${DIR}/_shared-vars.sh
 
-echo
-echo "+++ Check if changelog was updated +++"
```
Comment (PatrickSkowronek) [inline .travis/changelog.sh]: We use echo the show what will executed, even when there will nothing to be executed.

**Comment by PatrickSkowronek** [comment]
We have a convention in the shell files, which @danielkesselberg properly not really know? Please add it. The rest looks good, but I would test it locally.

File: .travis/changelog.sh (hunk 1 above)
Comment (danielkesselberg) [reply .travis/changelog.sh]: Not required here.

File: .travis/changelog.sh (hunk 2 above)
Comment (danielkesselberg) [reply .travis/changelog.sh]: https://github.com/geli-lms/geli/blob/60d2e6a8f6304fefd3fb5adcc18d2d4a67ce07ca/.travis/changelog.sh#L54-L55

Default case when a pull request. Every other case is covered.

File: .travis/changelog.sh (hunk 1 above)
Comment (aeimer) [reply .travis/changelog.sh]: Definetly on @PatrickSkowronek site! All shell files should start the same way, so we have everywhere the same "environment"

File: .travis/changelog.sh (hunk 2 above)
Comment (aeimer) [reply .travis/changelog.sh]: I'm on @PatrickSkowronek side, we always print the task we do on the first line. We should with this tradition. Makes sense in the travis log

File: .travis/changelog.sh [hunk 3]
```diff
@@ -2,43 +2,54 @@
...
-echo
+# shellcheck source=_shared-vars.sh
```
Comment (aeimer) [inline .travis/changelog.sh]: Please remove this comment

File: .travis/changelog.sh [hunk 4]
```diff
@@ -2,43 +2,54 @@
...
+. "${DIR}/_shared-vars.sh"
 
+if [[ ${TRAVIS_PULL_REQUEST} == false ]]; then
+  echo -e "${GREEN}+ Skip changelog check (no pull request)"
```
Comment (aeimer) [inline .travis/changelog.sh]: Always end with ${NC} if you define a color before

File: .travis/changelog.sh [hunk 5]
```diff
@@ -2,43 +2,54 @@
...
+  echo -e "${GREEN}+ Skip changelog check (dependency update)"
+  exit 0;
+fi
 
-if [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ "$TRAVIS_BRANCH" == "develop" ]; then
-  echo "+ detected pull request from ($TRAVIS_PULL_REQUEST_BRANCH) to $TRAVIS_BRANCH"
-  curl --silent https://raw.githubusercontent.com/geli-lms/geli/$TRAVIS_BRANCH/CHANGELOG.md \
+if [[ ${TRAVIS_BRANCH} == "develop" ]]; then
+  echo "+ Pull Request from ${TRAVIS_PULL_REQUEST_BRANCH} to ${TRAVIS_BRANCH}"
+  curl --silent https://raw.githubusercontent.com/geli-lms/geli/"${TRAVIS_BRANCH}"/CHANGELOG.md \
```
Comment (aeimer) [inline .travis/changelog.sh]: if you want to use " do it with the whole param: `curl --silent "https://raw.githubusercontent.com/geli-lms/geli/${TRAVIS_BRANCH}/CHANGELOG.md"`

File: .travis/changelog.sh [hunk 6]
```diff
@@ -2,43 +2,54 @@
...
+if [[ ${TRAVIS_BRANCH} == "master" ]]; then
+  echo "+ Pull Request from ${TRAVIS_PULL_REQUEST_BRANCH} to ${TRAVIS_BRANCH}"
+  curl --silent https://raw.githubusercontent.com/geli-lms/geli/"${TRAVIS_BRANCH}"/CHANGELOG.md \
   | diff CHANGELOG.md - \
   | grep -P '^< ## \[\[\d{1,3}\.\d{1,3}\.\d{1,3}\].*\] - \d{4}-\d{2}-\d{2} - .{10,}' - -q
+  RESULT=$?;
 
-  if [[ $? == 0 ]]; then
-    echo -e "${GREEN}+ Update in CHANGELOG.md found, exit${NC}"
+  if [[ ${RESULT} -eq 0 ]]; then
```
Comment (aeimer) [inline .travis/changelog.sh]: Why do you use the `-eq`?

File: .travis/changelog.sh (hunk 3 above)
Comment (danielkesselberg) [reply .travis/changelog.sh]: Annotation required by shellcheck

File: .travis/changelog.sh (hunk 6 above)
Comment (danielkesselberg) [reply .travis/changelog.sh]: Its common to use -eq to compare numbers

File: .travis/changelog.sh (hunk 3 above)
Comment (aeimer) [reply .travis/changelog.sh]: But it's not needed here. It's good that you check your code, but please remove it here. It just confuses...

**Comment by danielkesselberg** [comment]
@PatrickSkowronek @aeimer apart from your suggestions, you're seeing some problems with the functionality?

File: .travis/changelog.sh (hunk 3 above)
Comment (PatrickSkowronek) [reply .travis/changelog.sh]: Do you want to add this as a check? Otherwise can you explain why we need this.

Comment (danielkesselberg) [reply .travis/changelog.sh]: I would appreciate it but I'm not sure if I have enough time.

Comment (aeimer) [reply .travis/changelog.sh]: @danielkesselberg I don't understand what you want to say...

**Comment by aeimer** [comment]
@danielkesselberg yeah should work, you propably need to merge and try it...

**Comment by danielkesselberg** [comment]
image

**Pull request merged**